airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/connector_runner.py

```diff
@@ -168,3 +179,37 @@ def env_variables(self):
     @property
     def entry_point(self):
         return self._image.attrs["Config"]["Entrypoint"]
+
+    def _persist_new_configuration(self, new_configuration: dict, configuration_emitted_at: int) -> Optional[Path]:
+        """Store new configuration values to an updated_configurations subdir under the original configuration path.
+        N.B. The new configuration will not be stored if no configuration path was passed to the ConnectorRunner.
+        Args:
+            new_configuration (dict): The updated configuration
+            configuration_emitted_at (int): Timestamp at which the configuration was emitted (ms)
+
+        Returns:
+            Optional[Path]: The updated configuration path if it was persisted.
+        """
+        if self._connector_configuration_path is None:
+            logging.warning("No configuration path was passed to the ConnectorRunner. The new configuration was not persisted")
+            return None
+
+        with open(self._connector_configuration_path) as old_configuration_file:
+            old_configuration = json.load(old_configuration_file)
+
+        if new_configuration != old_configuration:
+            file_prefix = self._connector_configuration_path.stem.split("|")[0]
+            if "/updated_configurations/" not in str(self._connector_configuration_path):
```

Comment:
what's the story with updated_configurations? why do we use it?

Reply:
This folder will contain all the new configurations emitted by the connector. I decided to persist these in a subdirectory not to pollute the `secrets` root directory and allow quick cleaning (just erasing the `udpated_configurations` subdirectory) if needed.
The `connector_config_path_fixture` will glob in `updated_configurations` and use the latest created file if any, and will fall back to the root config in `secrets` otherwise.

Reply:
makes sense!

should we add it to `.gitignore` then when running the connector?

Reply:
`updated_configurations` will go under the original configuration path folder, which is usually the already gitignored `secrets` folder. But I will add it for safety.